tsdb/db.go

```diff
@@ -538,6 +538,15 @@ func Open(dir string, l log.Logger, r prometheus.Registerer, opts *Options) (db
 			segmentSize = opts.WALSegmentSize
 		}
 		wlog, err = wal.NewSize(l, r, filepath.Join(dir, "wal"), segmentSize, opts.WALCompression)
+		if errors.Cause(err).Error() == wal.ErrorUnsequentialSegments {
+			if err = wlog.Repair(errors.Cause(err)); err != nil {
```

Comment:
should increase the metric and also show a log for this repair, similiar to the other repair block, and for this the wal repair metric needs to be moved in the db struct
```
db.metrics.walCorruptionsTotal.Inc()
level.Warn(db.logger).Log("msg", "encountered WAL segment sequence error, attempting repair", "err", err)
```

Reply:
sure

Reply:
this is giving me `panic: runtime error: invalid memory address or nil pointer dereference`.